### PR TITLE
Erase vtk filename extension from the base name.

### PR DESCRIPTION
**Pull request by endJunction**
*Forgot to include this one yesterday into the NodePartitionedMesh PR.*

This makes a file name with ".vtk" extension a valid input for the partitioned mesh reader.

### CODE REVIEW

File: FileIO/MPI_IO/NodePartitionedMeshReader.cpp [hunk 1]
```diff
@@ -71,21 +72,25 @@ MeshLib::NodePartitionedMesh* NodePartitionedMeshReader::read(
 
     MeshLib::NodePartitionedMesh* mesh = nullptr;
 
+    /// Prefix of the filename of the partitioned mesh files.
+    std::string const filename_prefix =
+        boost::erase_last_copy(filename_prefix, ".vtk");
```
Comment (TomFischer) [inline FileIO/MPI_IO/NodePartitionedMeshReader.cpp]: The object `filename_prefix` is created here and at the same time used as an argument to the method. Is this working?

Comment (wenqing) [inline FileIO/MPI_IO/NodePartitionedMeshReader.cpp]: Not work, line 77 should be changed to
```
      boost::erase_last_copy(file_name_base, ".vtk");
```

Comment (wenqing) [inline FileIO/MPI_IO/NodePartitionedMeshReader.cpp]: @endJunction Tested with the above correction. You can amend the commit with that.

**Comment by endJunction** [comment]
Thanks for pointing out the mistake.
I have moved up the file extension handling, such that all of the extensions are handled in the same place, the `readMeshFromFile`.

@wenqing is it OK?

**Comment by endJunction** [comment]
BTW. This code is not being compiled; Why?

**Comment by wenqing** [comment]
:+1: .  All tests run. You can merge it.

The compilation error is caused by
```
#ifdef USE_PETSC
	for (auto p_it = project.processesBegin(); p_it != project.processesEnd(); ++p_it)
	{
		(*p_it)->releaseEquationMemory();
	}
```
in ogs,cpp, which should by in PR #580 , but it has been merged into the master vis  PR #831 (I have pointed out  this issue before in a comment). Anyway, there is not any test for ogs (#PETSC) executable file.

**Pull request merged**